lib/Mount/CollectiveFolderManager.php

```diff
@@ -194,14 +199,13 @@ public function getLandingPagePath(string $path, string $lang): string {
 	public function getFolderFileCache(int $id, string $name): array {
 		$qb = $this->connection->getQueryBuilder();
 		$qb->select(
-			'co.id AS folder_id', 'fileid', 'storage', 'path', 'fc.name AS name',
-			'mimetype', 'mimepart', 'size', 'mtime', 'storage_mtime', 'etag', 'encrypted', 'parent', 'fc.permissions AS permissions')
+			'co.id AS folder_id', 'fc.fileid', 'fc.storage', 'fc.path', 'fc.name AS name',
+			'fc.mimetype', 'fc.mimepart', 'fc.size', 'fc.mtime', 'fc.storage_mtime', 'fc.etag', 'fc.encrypted', 'fc.parent', 'fc.permissions AS permissions')
 			->from('collectives', 'co')
-			->leftJoin('co', 'filecache', 'fc', $qb->expr()->andX(
-				// concat with empty string to work around missing cast to string
-				$qb->expr()->eq('fc.name', $qb->func()->concat('co.id', $qb->expr()->literal(''))),
-				$qb->expr()->eq('parent', $qb->createNamedParameter($this->getRootFolderStorageId()))))
-			->where($qb->expr()->eq('co.id', $qb->createNamedParameter($id)));
+			->leftJoin('co', 'filecache', 'fc', $qb->expr()->eq('fc.name', $qb->expr()->castColumn('co.id', IQueryBuilder::PARAM_STR)))
+			->where($qb->expr()->eq('co.id', $qb->createNamedParameter($id)))
+			->andWhere($qb->expr()->eq('fc.parent', $qb->createNamedParameter($this->getFolderRootId())))
+			->andWhere($qb->expr()->eq('fc.storage', $qb->createNamedParameter($this->getRootFolderStorageId(), IQueryBuilder::PARAM_INT)));
```

Comment:
cannot judge whether specifying the storage id is safe here. Could there be, for example, incoming shares?